### PR TITLE
adding doc on Chainguard Terraform provider

### LINKED ISSUE #1307: Terraform provider introduction / how to
**What topic are you requesting a resource about?**
* **Chainguard product**
* Open source related
* Conceptual security related
* Other (please describe)

**Proposed title:**

How to use the Chainguard Terraform Provider

(link title: Terraform provider)

**Description:**

A general overview of our Terraform provider, based on [this draft blog post](https://docs.google.com/document/d/1tZxxfX4dS_FnRX9V6ZfqrK8F1HnAb-2VIwkJUpcF360/edit?usp=sharing). I think this might fit under the `Administration` tab but will decide on that later.

### PR DESCRIPTION
**Pull request by SharpRake**
## Type of change

This PR adds an introduction to the Chainguard Terraform provider. It's based on an as-yet-unpublished blog post from Colin.

I also made some small updates to the GH team rolebinding doc (referenced in the TF doc) due to changes in the provider. 

### What should this PR do?

Resolves #1307 

### How should this PR be tested?

This doc mostly just highlights how to use the TF provider without providing a procedural walkthrough of it. The examples are all valid but there's not a practical way to test them so just review for style and formatting.

### CODE REVIEW

File: content/chainguard/administration/terraform-provider/index.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+## Configuring the Chainguard Terraform provider
+
+Terraform uses [a native configuration language](https://developer.hashicorp.com/terraform/language) to define resources. Each configuration is stored in one or more `.tf` files, which in turn are stored in what is called a *root module*. When using the Terraform CLI, the root module is the working directory where you invoke `terraform` commands to create or destroy your resources.
+
+To use the Chainguard Terraform provider, add it to the block of required providers in your configuration:
+
+```
+terraform {
+  required_providers {
+	chainguard = {source = "chainguard-dev/chainguard"}
```
Comment (imjasonh) [inline content/chainguard/administration/terraform-provider/index.md:39]:
```suggestion
	chainguard = { source = "chainguard-dev/chainguard" }
```

Comment (SharpRake) [reply content/chainguard/administration/terraform-provider/index.md]: Thanks for flagging this, Jason!

File: content/chainguard/administration/terraform-provider/index.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+```
+terraform {
+  required_providers {
+	chainguard = { source = "chainguard-dev/chainguard" }
+  }
+}
+```
+
+If you don't have an active Chainguard token when you apply the configuration, the provider will automatically launch a browser to complete the Oauth2 flow with one of the default identity providers: GitHub, GitLab, or Google. This means the Terraform provider does not require `chainctl` to be installed to manage authentication with Chainguard.
```
Comment (ltagliaferri) [inline content/chainguard/administration/terraform-provider/index.md:44]: It looks like it's written like this in docs

```suggestion
If you don't have an active Chainguard token when you apply the configuration, the provider will automatically launch a browser to complete the OAuth 2 flow with one of the default identity providers: GitHub, GitLab, or Google. This means the Terraform provider does not require `chainctl` to be installed to manage authentication with Chainguard.
```

File: content/chainguard/administration/terraform-provider/index.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+You can customize the behavior of the authentication flow in a number of ways. For example, you can specify an identity to assume or a [verified organization](/chainguard/administration/iam-groups/verified-orgs/) name in order to use a previously-configured custom identity provider:
```
Comment (ltagliaferri) [inline content/chainguard/administration/terraform-provider/index.md:46]: nit: I would add the comma as they're discrete things

```suggestion
You can customize the behavior of the authentication flow in a number of ways. For example, you can specify an identity to assume, or a [verified organization](/chainguard/administration/iam-groups/verified-orgs/) name in order to use a previously-configured custom identity provider:
```

File: content/chainguard/administration/terraform-provider/index.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+}
+```
+
+You can find more information about authenticating with the Chainguard Terraform provider in the [provider documentation](https://registry.terraform.io/providers/chainguard-dev/chainguard/latest/docs) and [the included examples](https://github.com/chainguard-dev/terraform-provider-chainguard/tree/main/examples).
+
+## Defining Organization References
+
+Terraform was designed to allow users to manage various resources declaratively. In practice, this means that you define the state you want your resources to be in and then you let Terraform and the provider handle the details of bringing that state to reality. 
+
+Resources on the Chainguard platform are organized in a hierarchical structure of folder-like groups, with an organization's group at the root, though most users will only need to work at the organization level.
```
Comment (ltagliaferri) [inline content/chainguard/administration/terraform-provider/index.md:77]: Should it be "organizational level" or "`organization` level"?

File: content/chainguard/administration/terraform-provider/index.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+  login_options {
+	organization_name = "my-org.com"
+	# identity_id is the exact ID of an assumable identity.
+	# Get this ID with chainctl iam identities list
+	identity_id = "1f127a7c0609329f04b43d845cf80eea4247a07c/d6305475446bbef6"
+  }
+}
+```
+
+You can also configure the provider to use an OIDC token, either directly supplied or from a file, with the automatic browser flow disabled. This is useful when setting up CI workflows:
```
Comment (ltagliaferri) [inline content/chainguard/administration/terraform-provider/index.md:59]: nit: parallelism of the two options

```suggestion
You can also configure the provider to use an OIDC token, either by supplying it directly or pulling it from a file, with the automatic browser flow disabled. This is useful when setting up CI workflows:
```

File: content/chainguard/administration/terraform-provider/index.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+Resources on the Chainguard platform are organized in a hierarchical structure of folder-like groups, with an organization's group at the root, though most users will only need to work at the organization level.
+
+![Diagram outlining hierarchical structure of Chainguard resources. The diagram has two halves: one labeled "Organization" and another labeled "Chainguard". Under Organization is a box labeled "Tags" with an arrow pointing toward another box labeled "Repos." Under both halves is a box labeled "Role Bindings" which has four arrows pointing from it. Two arrows point to boxes (labeled "Identities" and "Custom Roles") under Organization and the other two point to boxes (labeled "User Identities" and "roles) under Chainguard.](tf-diagram.png)
+
+All user-managed resources are defined in relation to some parent group. This means developers need to be able to reference their organization's group throughout their configuration. We'll outline two ways to define this kind of reference: using local values and data sources.
```
Comment (ltagliaferri) [inline content/chainguard/administration/terraform-provider/index.md:81]: Maybe clarify this a bit if the parent group is always going to be the organization group

File: content/chainguard/administration/terraform-provider/index.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+
+```
+data "chainguard_group" "org" {
+  # This indicates the group is an organization.
+  parent_id = "/"
+  name  	= "[organization group name]"
+}
+```
+
+To refer to the organization's ID in other parts of your Terraform configuration, you would use the reference `data.chainguard_group.org.id`. The rest of the examples in this post will use this for referring to the organization's ID.
```
Comment (ltagliaferri) [inline content/chainguard/administration/terraform-provider/index.md:117]:
```suggestion
To refer to the organization's ID in other parts of your Terraform configuration, you would use the reference `data.chainguard_group.org.id`. The rest of the examples in this document will use this for referring to the organization's ID.
```

**Review by ltagliaferri — APPROVED** [review]
A few nits, feel free to merge when you are happy. Thank you for turning this around so quickly!

File: content/chainguard/administration/terraform-provider/index.md (hunk 6 above)
Comment (SharpRake) [reply content/chainguard/administration/terraform-provider/index.md:81]: added a note clarifying this point

File: content/chainguard/administration/terraform-provider/index.md (hunk 4 above)
Comment (SharpRake) [reply content/chainguard/administration/terraform-provider/index.md:77]: I don't think it should be code formatted, we're getting away from calling them "root groups" and moving to "organizations" and I think this is referencing that idea generally, not something out of a code example or something.

**Pull request merged**